packages/api/src/utils/db-client.js

```diff
@@ -554,19 +563,66 @@ export class DBClient {
         'uploads_multipart_total',
       ])
 
-    if (error) {
-      throw new DBError(error)
-    }
+    const dagByteSizeQuery = metricsQuery
+      .select('name, dimensions, value, collected_at')
+      .match({
+        name: 'dagcargo_project_bytes_in_active_deals',
+        dimensions: '{{project,nft.storage}}',
+      })
+      .order('collected_at', { ascending: true })
+      .single()
 
-    if (!data || !data.length) {
-      return undefined
-    }
+    const d = new Date()
+    d.setDate(d.getDate() - 7)
 
-    return data.reduce((obj, curr) => {
+    const dagByteSizeHistory = metricsLogQuery
+      .select('name, dimensions, value, collected_at')
+      .match({
+        name: 'dagcargo_project_bytes_in_active_deals',
+        dimensions: '{{project,nft.storage}}',
+      })
+      .lte('collected_at', d.toISOString())
+      .order('collected_at', { ascending: false })
+      .single()
+
+    try {
+      const fetchAllMetrics = await Promise.all([
+        primaryMetricQuery,
+        dagByteSizeQuery,
+        dagByteSizeHistory,
+      ])
+
+      const { data: primaryMetrics, error: primaryMetricsError } =
+        fetchAllMetrics[0]
+
+      const { data: dealsSize, error: dealsSizeError } = fetchAllMetrics[1]
+
+      const { data: dealsSizeHistory, error: dealsSizeHistoryError } =
+        fetchAllMetrics[2]
+
+      if (primaryMetricsError || dealsSizeHistoryError || dealsSizeError) {
+        // @ts-ignore
+        throw new DBError(error)
+      }
+
+      /** @type any  */
+      const outbound_data = {}
+
+      // Simple splatting of the metrics from first query
+      if (primaryMetrics && primaryMetrics.length) {
+        for (const metric of primaryMetrics) {
```

Comment:
I could also use .map to transform this, this felt better with the checks for existence than using object.keys.map.